leetcode/src/19.c

```diff
@@ -0,0 +1,32 @@
+/**
+ * Definition for singly-linked list.
+ * struct ListNode {
+ *     int val;
+ *     struct ListNode *next;
+ * };
+ */
+
+struct ListNode *removeNthFromEnd(struct ListNode *head, int n) {
+    struct ListNode *p_free, *p = head;
+    int i, sz = 0;
+    while (p != NULL) {
+        p = p->next;
+        sz++;
+    }
+    if (n == sz) {
+        p_free = head;
+        p = head->next;
+        free(p_free);
+        return p;
+    }
+    for (i = 0, p = head; i < sz - n - 1; i++, p = p -> next)
+    ;
```

Comment:
This `for` with an empty body surely looks scary, while it does it's job just fine, you should probably put `p = p -> next` in the loop body, for readability's sake if nothing else. also, if `i` is only used here as the loop variable, why not declare it here instead?
```suggestion
    for (int i = 0, p = head; i < sz - n - 1; i++) {
        p = p -> next;
    }
```

Reply:
You cannot declare variable inside for loop fields in C (unlike C++). For the pointer increment inside for loop, it is more of a style taste, happy to change if more reviewers feel the same way as you do, thanks

Reply:
Oh, I have no intention to force anything on you, sorry if my comment came out as more blunt than I had intended it to be, but I'm pretty sure `for(int i = 0; i < n; i++)` is a thing in C

Reply:
Sorry, yes you are right, it was introduced in C99 (https://stackoverflow.com/questions/1287863/c-for-loop-int-initial-declaration), I was thinking about the earlier standards :) Nevertheless, if putting "int" inside the for initialization field, would need to move "p = head" outside the for() initialization field, thanks